project/src/main/webapp/profile.js

```diff
@@ -1,14 +1,62 @@
 $(document).ready(function() {
-  fetch('/profile-update')
-      .then((response) => response.json())
-      .then((userData) => {
-        const userName = userData['name'];
-        const userEmail = userData['email'];
-        const userInterests = userData['interests'];
-        const userSkills = userData['skills'];
-        $('#name').text(userName);
-        $('#email').text(userEmail);
-        $('#interests').text(userInterests);
-        $('#skills').text(userSkills);
-      });
+  updateProfile();
 });
+
+async function updateProfile() {
+  const userData = await getCurrentProfileData();
+  console.log(userData);
+  updateProfileBasics(userData);
+  updateProfileEvents();
+}
+
+/** Get all information for current user from backend */
+async function getCurrentProfileData() {
+  const userData = await fetch('/profile-update');
+  return userData.json();
+}
+
+/** Populate the user profile with name, email, interests, skills */
+function updateProfileBasics(userData) {
+  const userName = userData['name'];
+  const userEmail = userData['email'];
+  const userInterests = userData['interests'];
+  const userSkills = userData['skills'];
+  $('#name').text(userName);
+  $('#email').text(userEmail);
+  $('#interests').text(userInterests);
+  $('#skills').text(userSkills);
```

Comment:
Stylewise would it be good to not declare the const, and inline? 
eg.  $('#name').text( userData['name']); but this could just be personal preference

Reply:
Yeah it's probably readable enough without two lines, I'll fix it